### PR TITLE
chore(override): add Noop Overrider

### PR DESCRIPTION
**Pull request by efekarakus**
Related #4208 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/override/noop.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package override
+
+// Noop represents an Overrider that does not do any transformations.
+type Noop struct{}
```
Comment (iamhopaul123) [inline internal/pkg/override/noop.go:7]: Why do we need this 🤔

Comment (efekarakus) [reply internal/pkg/override/noop.go:7]: The idea is to be able to do something like this from a client:

```go
info, err := override.Lookup(path/to/overrides)
if err != nil {
   return new(override.Noop)
}
if info.IsCDK() {
   return override.WithCDK(info.Path())
}
return override.WithYAMLPatch(info.Path())
```

This way the `deploy` package doesn't need to do any  `if override != nil` check and can just call `overrides.Override`

Comment (iamhopaul123) [reply internal/pkg/override/noop.go:7]: but why not return an error if `err != nil` instead of creating a `new(override.Noop)`?

Comment (efekarakus) [reply internal/pkg/override/noop.go:7]: Here is a sample scenario to illustrate:
```
copilot/
  frontend/
      manifest.yml
```
There is no `overrides/` directory defined by the customer, so we don't want to error.

My plan is to have a flow like this:
```go
info, err := override.Lookup("copilot/frontend/overrides")
if err != nil {
   if errors.Is(err, override.ErrNotExist) { return new(override.Noop), nil } // This is fine, have an overrider that doesn't do anythign.
   return nil, err
}
if info.IsCDK() {
  return override.WithCDK(info.Path()), nil
}
return override.WithYAMLPatch(info.Path()), nil
```

**Review by iamhopaul123 — APPROVED** [review]

**Review by huanjani — APPROVED** [review]
🚢

**Pull request merged**